packages/zero-cache/src/services/view-syncer/client-handler.ts

```diff
@@ -263,19 +273,28 @@ const lmidRowSchema = v.object({
   lastMutationID: v.number(), // Actually returned as a bigint, but converted by ensureSafeJSON().
 });
 
-function makeRowPatch(patch: RowPatch): RowPatchOp {
+function makeRowPatch(patch: RowPatch, protocolVersion: number): RowPatchOp {
   const {
     op,
     id: {table: tableName, rowKey: id},
   } = patch;
 
   switch (op) {
-    case 'put':
+    case 'put': {
+      const value = v.parse(ensureSafeJSON(patch.contents), rowSchema);
+      const contents =
+        protocolVersion >= 4
```

Comment:
this protocolVersion is a property of the connection and set by the client, correct?